src/tools/debugger/entities/SteppingType.java

```diff
@@ -30,44 +37,86 @@ public void process(final SuspendedEvent event) {
-  STEP_TO_NEXT_MESSAGE("todo", "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final SuspendedEvent event) { /* TODO */ } },
-  STEP_RETURN_TO_PROMISE_RESOLUTION("todo", "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final SuspendedEvent event) { /* TODO */ } };
+  @SerializedName("stepIntoActivity")
+  STEP_INTO_ACTIVITY("stepIntoActivity", "Step into Activity", Group.ACTIVITY_STEPPING, "arrow-down", new Class[] {ActivityCreation.class}) {
+    @Override
+    public void process(final Suspension susp) {
+      // assert SOMNode.unwrapIfNecessary(susp.getEvent().getNode()) instanceof SpawnPrim || SOMNode.unwrapIfNecessary(susp.getEvent().getNode()) instanceof SpawnWithPrim;
+      susp.getEvent().prepareStepOver(1);
+      susp.getActivityThread().setSteppingStrategy(new IntoSpawn());
+    }
+  },
+
+  @SerializedName("returnFromActivity")
+  RETURN_FROM_ACTIVITY("returnFromActivity", "Return from Activity", Group.ACTIVITY_STEPPING, "arrow-left",
+      null, new ActivityType[] {ActivityType.PROCESS, ActivityType.TASK, ActivityType.THREAD}) {
+    @Override
+    public void process(final Suspension susp) {
+      susp.getEvent().prepareContinue();
+      susp.getActivityThread().setSteppingStrategy(new ReturnFromActivity());
+    }
+  },
+
+
+  @SerializedName("stepToChannelRcvr")
+  STEP_TO_CHANNEL_RCVR("stepToChannelRcvr", "Step to Receiver", Group.PROCESS_STEPPING, "arrow-right", new Class[] {ChannelWrite.class}) {
+    @Override
+    public void process(final Suspension susp) {
+      susp.getEvent().prepareStepOver(1);
+      susp.getActivityThread().setSteppingStrategy(new ToChannelOpposite());
+    }
+  },
+
+  @SerializedName("stepToChannelSender")
+  STEP_TO_CHANNEL_SENDER("stepToChannelSender", "Step to Sender", Group.PROCESS_STEPPING, "arrow-left", new Class[] {ChannelRead.class}) {
+    @Override
+    public void process(final Suspension susp) {
+      susp.getEvent().prepareStepInto(1);
+      susp.getActivityThread().setSteppingStrategy(new ToChannelOpposite());
+    }
+  },
+
+
+  STEP_TO_RECEIVER_MESSAGE("todo",   "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final Suspension susp) { /* TODO */ } },
+  STEP_TO_PROMISE_RESOLUTION("todo", "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final Suspension susp) { /* TODO */ } },
+  STEP_TO_NEXT_MESSAGE("todo",       "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final Suspension susp) { /* TODO */ } },
+  STEP_RETURN_TO_PROMISE_RESOLUTION("todo", "todo", Group.ACTOR_STEPPING, "arrow-right", null) { @Override public void process(final Suspension susp) { /* TODO */ } };
 
   public enum Group {
     BASIC_CONTROLS("Basic Controls"),
     LOCAL_STEPPING("Local Stepping"),
-    ACTOR_STEPPING("Actor Stepping");
+    ACTIVITY_STEPPING("Activity Stepping"),
```

Comment:
Activity Stepping is more general than Actor Stepping.
It local to one or two activities, step into activity and return from activity. Please see where `ACTIVITY_STEPPING` is actually used.